cwl/src/main/scala/cwl/ExpressionEvaluator.scala

```diff
@@ -24,6 +25,16 @@ object ExpressionEvaluator {
   type ECMAScriptFunction = String Refined MatchesRegex[ECMAScriptFunctionWitness.T]
   type MatchesECMAFunction = MatchesRegex[ECMAScriptFunctionWitness.T]
 
+  // This is not ECMAScript, just what CWL uses for interpolated expressions. A dollar sign immediately followed
+  // by a paren followed by anything that isn't a close paren followed by a close paren. Capturing groups around
+  // the fore, expression and after portions to concatenate fragments for evaluation. Obviously this will break
+  // with nested parens. It's not clear what the specification is here; this simplistic implementation may need
```

Comment:
Please ask & verify what they think the spec means before merging

Reply:
I reviewed the spec to formulate a question but it turns out I need to revise this comment since the spec is actually [quite clear](http://www.commonwl.org/v1.0/CommandLineTool.html#Parameter_references) on this. What I'm not sure about is if our expression evaluation currently does that "Objects entries are sorted by key" bit already. I can pick the relevant people's brains on this next week.

Reply:
Sounds good. I thought it was pretty straightforward on their side and that you had it right but my spec-fu was failing earlier tonight.